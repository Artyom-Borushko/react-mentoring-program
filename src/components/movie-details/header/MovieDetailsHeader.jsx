import '../../../styles/movieDetailsHeader.css';

export default function MovieDetailsHeader({searchAllHandler}) {

    return (
        <div className={'movie-details-header'}>
            <p className={'movie-details-header-text'}>netflixroulette</p>
            <img
                src={'/images/buttons/searchButton.png'}
                alt='search icon'
                className={'movie-details-search-icon'}
                aria-label={'search icon'}
                onClick={searchAllHandler}
            />
        </div>
    );
}
