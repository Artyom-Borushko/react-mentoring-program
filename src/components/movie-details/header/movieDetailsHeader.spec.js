import {render, screen} from "@testing-library/react";
import React from "react";
import '@testing-library/jest-dom';
import userEvent from "@testing-library/user-event";
import MovieDetailsHeader from "./MovieDetailsHeader";

describe("MovieDetailsHeader component tests", () => {

    it("renders MovieDetailsHeader component", () => {
        const {container} = render(<MovieDetailsHeader/>);

        expect(container).toMatchSnapshot();
    });

    it("search image is clickable", async () => {
        const mockClickHandler = jest.fn();
        render(<MovieDetailsHeader
            searchAllHandler={mockClickHandler}
        />);

        await userEvent.click(screen.getByRole('img', {name: 'search icon'}));

        expect(mockClickHandler).toHaveBeenCalledTimes(1);
    });
});
