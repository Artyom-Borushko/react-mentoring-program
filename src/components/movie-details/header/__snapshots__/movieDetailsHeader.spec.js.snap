// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetailsHeader component tests renders MovieDetailsHeader component 1`] = `
<div>
  <div
    class="movie-details-header"
  >
    <p
      class="movie-details-header-text"
    >
      netflixroulette
    </p>
    <img
      alt="search icon"
      aria-label="search icon"
      class="movie-details-search-icon"
      src="/images/buttons/searchButton.png"
    />
  </div>
</div>
`;
